TypeCobol/AntlrGrammar/Cobol2002CodeElements.g4

```diff
@@ -44,6 +45,13 @@ initializeStatement:
 	(THEN? REPLACING initializeReplacingDirective+)?
 	(THEN? TO? DEFAULT)?;
 
+// New Cobol v6.1 ALLOCATE statement to obtain dynamic storage.
+// 'INITIALIZED' is defined here as a contextual keyword. The dataNameReference is therefore not allowed to be named 'INITIALIZED' in this statement.
+allocateStatement:
+	ALLOCATE ((arithmeticExpression CHARACTERS) | { !string.Equals(CurrentToken.Text, "INITIALIZED", System.StringComparison.InvariantCultureIgnoreCase) }? dataNameReference)
+	({ string.Equals(CurrentToken.Text, "INITIALIZED", System.StringComparison.InvariantCultureIgnoreCase) }? KeywordINITIALIZED=UserDefinedWord)?
+	(RETURNING dataItemReference)?;
```

Comment:
Use variable2 here.
The difference with dataItemReference is that it's mapped to Variable class.

Then use CoboLExpressionsBuilder.CreateVariable to instantiate Variable.